pkgs/by-name/wl/wl-shimeji/package.nix

```
@@ -0,0 +1,76 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  makeWrapper,
+  pkg-config,
+  which,
+  python3,
+  wayland,
+  wayland-scanner,
+  wayland-protocols,
+  libarchive,
+  uthash,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wl-shimeji";
+  version = "0.0.2";
+
+  src = fetchFromGitHub {
+    owner = "CluelessCatBurger";
+    repo = "wl_shimeji";
+    rev = "bf0150a313c3df271d249d3e14a17b48193f67e8";
+    hash = "sha256-ag3MHyyfo+2yZxPSBZgBsVZ4a6nLSv1KMUcYS34lIiM=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    wayland-protocols
+    wayland-scanner
+    libarchive
+    uthash
+    pkg-config
+    which
+    python3
+    makeWrapper
+  ];
+  buildInputs = [
+    wayland
+    libarchive
+  ];
```

if you enable `strictDeps`, you'll see that you get some of these wrong: `wayland*` and Libarchive are all used as libraries, not as executables
```suggestion
  nativeBuildInputs = [
    pkg-config
    which
    python3
  ];
  buildInputs = [
    wayland
    wayland-protocols
    wayland-scanner
    uthash
    libarchive
    (python3.withPackages (ps: [ ps.pillow ]))
  ];
```
+ Python env in `buildInputs` will be needed by `patchShebangs`, so you don't need `makeWrapper`